In [13]:
import re
import nltk
from nltk.corpus import stopwords
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.metrics.pairwise import cosine_similarity
import pandas as pd

nltk.download('stopwords')

text1 = """Everyone is a manager. People learn how to manage their lives and their relationships with others. They learn how to organize their time and work. In business, management is usually a team of people who operate the business and try to get the work done. They organize the time and work, and they manage relationships among the employees who work together.
Management has five main jobs: to plan, direct, control, coordinate, and organize. Managers at different levels spend different amounts of time on these jobs. For instance, a president is a manager who spends a lot of time on planning. A supervisor is a manager who spends more time on directing.
The most important job of management is planning the goals of a business and how these goals will be completed. In planning, managers must consider five variables: personnel, money, machines, materials and methods. There are four steps in the planning process. First, the goals must be identified and defined. For example, the managers may decide to increase production by 10 percent in two years. Then the managers must select the best plan for completing the goal. Second, the managers must get approval for the plan. Third, they must check its effectiveness. Is it working? Fourth, they must be able to change the plan if it is not working.
Management also directs and supervises. Managers usually know their workers well so that they can lead and give instructions. Managers must communicate clearly with workers so that they will understand their jobs and responsibilities. Good managers get the cooperation of all their workers. Directing is a difficult job for managers, but good supervision of workers is essential for business success.
Managers control too. They decide on the quality of the work or the products. Is the work good? If the quality is low, the manager looks for the reasons. Managers decide if the workers are fast or slow. Are the workers good or bad? Are there enough workers? Is the plan working? If necessary, the managers hire more workers or make changes in the company policy. Managers control the workers, time, materials, and the job.
Another job of managers is to coordinate. All parts of an organization need to work together. Workers must understand their jobs and how to do them efficiently. When there are problems, the manager must solve them. Good managers must communicate well to coordinate goals and plans. All managers at every level must coordinate well in order to keep an organization together.
The fifth job of management is to organize. All businesses have a basic structure which determines the power and authority of each level of the organization. For example, a company may have only one vice-president or it mat have several who are in charge of separate departments such as production, marketing, or finance. Within each department, there are managers who organize the workers and their jobs to increase efficiency. All workers know their jobs and their bosses.
The key to success in the business world is good management."""
text2 = """The role of shipping in the economy as we have seen has changed from being an industry of national importance or military support to being a driver of economic growth and a medium to balance the trade deficit. Shipping industry has also been used as a political and economic instrument to establish trade relations with certain countries or a specific region. Not only do such trade relations help countries leverage each others low price products and establish political relations. But as shipping has become more commercialised corporate interest in its operations has grown. The governments have also realised the need to improve infrastructure.
In May 2008, the World Shipping Council released a statement about rising costs of bunker fuel for freighters:
“Shipping lines worldwide are struggling as crude oil prices topped an unprecedented U.S.$119 per barrel this week, in turn pushing marine bunker fuel prices up past $552 per ton – a $26 per ton increase since the end of March alone.  Bunker prices have risen 87% since the beginning of 2007.
Fuel costs represent as much as 50-60% of total ship operating costs, depending on the type of ship and service.   Ocean carriers are required to recover these costs to maintain levels of service, meaning the price of shipping goods will continue to face upward pressures.”
In such a situation the government’s intervention becomes highly important. The governments can try and convince OPEC to increase supply thereby reducing the cost of crude oil in the futures market. This would allow shipping companies to hedge their position for future. Governments also levy their own import and export duties along with the tax on fuel for ships. Relaxation in these fields or temporary measures such as abolishment of taxes would also help the industry. It is only the government that can take such a call. As it is in the position to look at the shipping industry from a holistic perception of national interest rather than a problem concerned with a specific industry. Hence, the role of the government becomes important from an administrative purpose.
Most economies especially in the developing world have been closed economies. Domestic private ownership has been limited in the shipping industry and international investment has been negligible. With the opening up of the economies the government also needs to develop policies and legal frameworks by which the role of domestic and international players in the shipping industry can be defined, facilitated and monitored. All these are tasks that can be undertaken by the government only. These tasks should fall under the national shipping policy.
But, the role of the government in such matters becomes important as highlighted by the 2006 crisis that followed the Dubai ports authorities’ ability to control up to 6 ports in the United States. There was an attempt in 2006 to approve the sale of  one of the largest U.S. companies in the shipping industry. The Committee on Foreign Investment in the United States (CFIUS) approved a deal that would allow a company owned by the government of United Arab Emirates (UAE) to purchase the Peninsular & Oriental Steam Navigation Company (P&O). The deal raised a lot of controversy in the American political circles and media.
The deal would mean that terminal operations at six U.S. ports would have been controlled by the new company. This would give the company the strategic control of the facilities at the ports. The deal bought forth objections on two fronts. The role of the foreign investment in strategic shipping infrastructure and the security threat to the integrity of the U.S. were the two divisive factors amongst those supporting and against the deals. Memories of 26/11 hijackings were bought forth with the fact that the hijackers used Dubai as a transit point and that two of them were citizens of UAE. The new owners had to finally cave in and had to sell their stake back to an American company.
This highlights the important role that governments till today have to play to define and manage the implementation of shipping policy. If the consensus view in U.S. is that only domestic companies can control terminal operations at domestic ports then this criteria should be part of the national shipping policy. Any changes to that must be preceded by administrative procedures such as discussions in the Congress and legal changes could be made by asking the Congress to support suitable legislations.
"""
text3 = """On the Isles of Scilly, the fly in for the summer, you can picnic on rocks accompanied only by seals, and you go about not by bus but by boat. It's a long way from the traffic- choked streets of London, and it's not difficult to see why Toby Tobin-Dougan swapped one for the other. Now when he's working he looks out on a field of flowers rather than a road full of taxis. Toby and his wife Louise live on one of the larger islands, St Martin's, and run a bakery of the same name, supplying bread to their own island and delivering by boat to their neighbors on the other inhabited islands of Tresco, Bryher, St Agnes, and St Mary's. It sounds idyllic, and in many ways it is. But it's also, hard work, although Toby and Louise are now helped by a night baker who produces most of the basic breads overnight. They bake on Mondays, Wednesdays, and Fridays in the winter, and every day including VES Sunday during the summer. On baking days they have to be up at four in the morning to make the dough. They shape all their loaves by hand, finishing the baking by mid-morning when they prepare the deliveries and work in the shop, where the queues often stretch out of the door.
So what inspired them to leave the rat race and move to one of the mildest climates and most beautiful environments in Britain? Toby, 44, first visited St Martin's in the early 1980s and was irresistibly drawn to the island. At the time he was living in Brighton but working in London, running a photographic laboratory. He finally sold the business in 1992, packed his bags, and moved to St Martin's. He found accommodation with a local fisherman and worked on boats and in the St Martin's Hotel. Louise, 34, from Bristol, also fell in love with the island during a holiday and found a job in the same hotel. They have been married for five years and both knew that they wanted to start their own business. Breadmaking happened almost by chance – the learning process was very much atrial and error. “We taught ourselves to bake in our own kitchen,'' Toby remembers. "We sold our loaves at the campsite nearby for campers to buy – just 25 or 30 a day.” The business expanded rapidly. The bakery opened four years ago and this year it will be extended into the old fire station next door. The shop is becoming more like a deli, offering picnic food for holidaymakers to take to the beach. Toby and Louise now grow their own organic fruit and vegetables and keep ducks, chickens, and a pig. They have also started running baking classes in the quieter months of the year. The point of all their work, Toby says, isn't money. They both earn less than they would have if they'd stayed in hotel work. The satisfaction comes from providing a useful service – and living in a place for which they have a real 'love and affinity'. Is there anything they miss from their pre-Scillies life? As their two dogs wander happily around and a falcon flies overhead in the fresh wind of the sea, there is a long silence.
"""
text4 = """Engaging in a combination of high control methods of market entry, they have since grown to become the country’s most successful fast food company, with 4563 stores and 11.6% of the market share (Euromonitor, 2018). Control, the capacity to impact procedures, strategies and systems (Anderson and Gatignon 1986), will be discussed in relation to these/their/the market entry methods using takeaways from TCA framework adapted from Ronald Coase (1937). It will BE ARGUED THAT control is critical to the successful internationalisation of a firm (Anderson and Gatignon 1986) and KFC’s desire for control led them to employ/ opt for a series of methods that enabled their rapid expansion, quality control and consequently renowned success.
However, it will be DEBATED(FIND OTHER WORD) that operations with a high degree of control result in inflexibility (Holton o1971); this will be framed regarding the threats facing the firm, and the extent to which this inflexibility ultimately led to KFC’s exit from the Chinese market with the creation of YUM China in 2016
To discuss KFC’S entrance in to China, it is beneficial to evaluate the motivating forces behind the expansion and their choice of FDI. The most widely recognized theory for FDI is Dunning’s Eclectic Paradigm (Johns, 2018), which attempts to explain the pattern and MAGNITUDE of international production (Dunning, 1976). Even though it fails to take into account some of the broader market features (Johanson and Matsson 1986) it explores the conditions required by the firm for internationalisation through Foreign Direct Investment.
Dunning (1976) defines three possible sources of advantage to the firm; Ownership, Location and Internalisation. The first involves specific advantages unique to the firm that affords them a “net competitive advantage”(rugman 2010) that suggests why a firm may succeed abroad; KFC benefitted from the intangible asset of a reputation, that is easily transferrable to the proposed subsidiary (Gillies, 2005) , which allowed them to be seen as reputable luxury western product in china, at a time when China was looking to the West for inspiration. Having vast financial capital afforded to them by their parent company YUM! they could exploit the opportunity ahead of their competitors.
Location seeks to explain which market a firm chooses to enter; the size of the Chinese market, with 20% of the words population, and an average GDP growth of 10% (Data.worldbank.org, 2019) suggest KFC were looking to tap into foreign demand (Rugman and Collinson 2012). Together with the relative lack of competition they faced, China certainly held location advantages for KFC.
Finally Internalisation refers to how the company structures its operations; the degree to which they internalize their value chain in the host country. Dunning asserts that it must be more profitable for the firm to internalize than outsource (Dunning, 1988). According to Eithier internalisation is “emerging as the caesar of the triumvirate” (Ethier, 1986: 803 cited in Dunning, 1998) suggesting the degree to which the firm decides to internalize its production is the most influential/vital aspect of th the paradigm. This was ceetainly true for KFC as it dictated their level of embeddedness in China; due to their desire for control they had a fully internalized value chain in China.
"""

texts = [text1, text2, text3, text4]
titles = ["Текст 1", "Текст 2", "Текст 3", "Текст 4"]

for i, text in enumerate(texts):
    print(f'Длина текста {i + 1}: {len(text)}')

Длина текста 1: 3030
Длина текста 2: 4476
Длина текста 3: 2981
Длина текста 4: 3355


[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [11]:
def preprocess(text):
    text = text.lower()
    text = re.sub(r'[^а-яёa-z\s]', ' ', text)
    text = re.sub(r'\s+', ' ', text).strip()
    stop_words = set(stopwords.words('russian'))
    words = [w for w in text.split() if w not in stop_words]
    return ' '.join(words)

clean_texts = [preprocess(t) for t in texts]
vectorizer = TfidfVectorizer()
tfidf_matrix = vectorizer.fit_transform(clean_texts)
sim_matrix = cosine_similarity(tfidf_matrix)
df_sim = pd.DataFrame(sim_matrix, index=titles, columns=titles)
print("Косинусное сходство между текстами:")
print(df_sim.round(3))

Косинусное сходство между текстами:
         Текст 1  Текст 2  Текст 3  Текст 4
Текст 1    1.000    0.482    0.520    0.477
Текст 2    0.482    1.000    0.588    0.597
Текст 3    0.520    0.588    1.000    0.560
Текст 4    0.477    0.597    0.560    1.000


Я ожидала, что наиболее близкими окажутся тексты 2 и 4, поскольку оба посвящены экономико‑политической проблематике. Этот прогноз подтвердился: пара Текст 2 и Текст 4 получила максимальное косинусное сходство 0.597. Второе место заняла пара Текст 2 и Текст 3 – такой результат уже менее очевиден, ведь Текст 3 лирически описывает быт пекарей на островах Силли и далёк от макроэкономики. Удивило и то, что Текст 1 («менеджмент») оказался наиболее близок именно к «пекарне» (Текст 3, 0.520), а не к деловым текстам 2 и 4 (с ними сходство всего 0.482 и 0.477).

Причина такого распределения – особенность TF-IDF векторизации на униграммах. Удалив стоп‑слова, мы всё равно оставляем широкий пласт общеупотребительной лексики, связанной с работой и бизнесом: work, business, time, company, shop, provide. В «менеджменте» и «пекарне» эти слова повторяются регулярно (планирование, работа, магазин, обслуживание), что создаёт иллюзию тематической близости. Тексты 2 и 4, напротив, перенасыщены специализированной терминологией (shipping, bunker fuel, government, FDI, Dunning, internalisation), которая почти не пересекается с лексикой общего менеджмента из Текста 1, отсюда и низкие значения. TF-IDF фиксирует не смысловую, а чисто лексическую общность, и в данном корпусе именно повседневная «рабочая» лексика оказалась сильным связующим фактором, иногда перевешивающим тематическое сходство.